### ISSUE TITLE
Overriding types from "@hotwired/stimulus"

### ISSUE THREAD

**Issue by YummYume** [lhapaipai/symfony-vite-dev#36]
### vite-plugin-symfony version

6.5.0

### vite-bundle version

6.5.0

### your OS, node version, Symfony version, PHP version

Node 20.13.1, TypeScript 5.3.3

### Description

When upgrading to more recent versions of TypeScript (> 5.3), we started experiencing errors such as `Controller is not generic`. By digging a little, we found that the root cause is the definition from this bundle that is causing a full override of the types from `@hotwired/stimulus`. This is the [concerned line](https://github.com/lhapaipai/vite-plugin-symfony/blob/4ac46e7c2c505eda50243877cdbcd81c17876e23/src/stimulus/helpers/base.ts#L6).

### How to reproduce

I'm not sure if it's tied to a specific TypeScript version, but having this bundle with the latest TypeScript version causes types from `@hotwired/stimulus` to be overridden in any controller. This is what the `tsconfig.json` file looks like :

```json
{
    "compilerOptions": {
        "target": "ESNext",
        "module": "ESNext",
        "moduleResolution": "Bundler",
        "strict": true,
        "esModuleInterop": true,
        "forceConsistentCasingInFileNames": true,
        "allowJs": true,
        "checkJs": false,
        "outDir": "/public/dist/",
        "noImplicitAny": true,
        "baseUrl": ".",
        "skipLibCheck": true,
        "isolatedModules": true,
        "types": ["vite/client", "@types/cloudflare-turnstile", "@types/hotwired__turbo"],
        "paths": {
            // ...
        }
    },
    "include": [
        // ...
    ],
    "exclude": [
        // ...
    ]
}
```

### Possible Solution

I did very little testing, but by using `interface Controller` instead of `export class Controller`, the issue seemed to be resolved, and the 2 definitions are now taken into account (merged) by TypeScript. It would be either this, or make sure the definitions declared from this bundle stay within the bundle and do not impact code outside of it. What do you think ?

**Comment by lhapaipai** [comment]
hi @YummYume,
thanks for your time.

I followed your advice and you can find a corrected version with vite-plugin-symfony 6.5.1

I'm not very comfortable with merging modules into other modules
but now
the declaration file seems clean

```ts
// vite-plugin-symfony/dist/stimulus/helpers/index.d.ts
import { Application } from '@hotwired/stimulus';
import { I as ImportedModules, C as ControllerModule } from '../../types.d-o_j0yKco.js';
import 'react';
import 'vue';
import 'svelte';

declare module "@hotwired/stimulus" {
    interface Controller {
        __stimulusLazyController: boolean;
    }
}
declare function startStimulusApp(): Application;
declare function registerControllers(app: Application, modules: ImportedModules<ControllerModule>): void;

export { registerControllers, startStimulusApp };
```
I let you test and close the issue if that solved your problem otherwise let me know and I will look for a way to not have the merge declaration appear in the `vite-plugin-symfony/dist/stimulus/helpers/index.d.ts` file.

have a good day.

**Comment by YummYume** [comment]
Hi @lhapaipai, thanks for addressing this !

I just tested with `vite-plugin-symfony@6.5.1` and everything seems to work again, both definitions are now merged and available ! Closing this :+1:
